plugins/catalog-backend-module-bitbucket-server/src/providers/BitbucketServerEntityProvider.ts

```diff
@@ -213,4 +240,332 @@ export class BitbucketServerEntityProvider implements EntityProvider {
     }
     return result;
   }
+
+  /**
+   * Checks if the webhook was triggered on a commit to the head branch of a repository
+   * @param event Bitbucket Server webhook repo:refs_changed event
+   */
+  private isHeadPush(event: Events.PushEvent): boolean {
+    return event.changes.some(c => DEFAULT_BRANCH.includes(c.ref.displayId));
```

Comment:
`DEFAULT_BRANCH[ES]` includes master and main. Git itself still defaults to master, and many moved to use main instead. You may use something else like "production" or "release" or whatever.

This includes the assumption that one and only one of these two branch names is used.

The need for this seems to come from the use of the event `repo:refs_changed` which I assume is not limited to the default branch only.
Other systems provide events for pushes to the default branch. Is there no alternative event like that?

Alternatively, it might be needed to allow users to configure the list of default branch names that they consider. The issue of false positives would still remain then (e.g., one repo using "release" as default branch, but another one using "main" and having additionally a "release" branch).

Reply:
When I was building this event system there is no determining factor to tell from an event payload that the event is coming from a push to the default branch of a repository. This is the only event payload and the only way to know if it is the head branch is to search for it in the changes section of the event payload. I can include some options in the config to allow for this to be editable.

Reply:
I understand, thanks! You could keep this for a follow-up as well.